tests/sentry/incidents/endpoints/test_project_alert_rule_index.py

```diff
@@ -343,9 +306,115 @@ def test_async_lookup_outside_transaction(self, mock_uuid4, mock_get_channel_id)
         with self.feature("organizations:incidents"), self.tasks():
             resp = self.get_response(self.organization.slug, self.project.slug, **test_params)
         assert resp.status_code == 400
-        assert (
-            mock_get_channel_id.call_count == 3
-        )  # Did not increment from the last assertion because we early out on the validation error
+        # Did not increment from the last assertion because we early out on the validation error
+        assert mock_get_channel_id.call_count == 3
+
+    @responses.activate
+    def test_success_response_from_sentry_app(self):
+        responses.add(
+            method=responses.POST,
+            url="https://example.com/sentry/alert-rule",
+            status=202,
+        )
+
+        sentry_app = self.create_sentry_app(
+            name="foo",
+            organization=self.organization,
+            schema={
+                "elements": [
+                    self.create_alert_rule_action_schema(),
+                ]
+            },
+        )
+        install = self.create_sentry_app_installation(
+            slug="foo", organization=self.organization, user=self.user
+        )
+
+        sentry_app_settings = [
+            {"name": "title", "value": "test title"},
+            {"name": "description", "value": "test description"},
+        ]
+
+        alert_rule = {
+            **self.valid_alert_rule,
+            "triggers": [
+                {
+                    "actions": [
+                        {
+                            "type": "sentry_app",
+                            "targetType": "sentry_app",
+                            "targetIdentifier": sentry_app.id,
+                            "hasSchemaFormConfig": True,
+                            "sentryAppId": sentry_app.id,
+                            "sentryAppInstallationUuid": install.uuid,
+                            "settings": sentry_app_settings,
+                        }
+                    ],
+                    "alertThreshold": 300,
+                    "label": "critical",
+                }
+            ],
+        }
+
+        with self.feature(["organizations:incidents", "organizations:performance-view"]):
+            self.get_valid_response(
+                self.organization.slug, self.project.slug, status_code=201, **alert_rule
+            )
+
+    @responses.activate
+    def test_error_response_from_sentry_app(self):
```

Comment:
This tests that the error message from Sentry Apps will be surfaced for metric alerts (POST endpoint)